app/src/main/java/com/yacgroup/yacguide/LaunchActivity.kt

```diff
@@ -84,19 +86,92 @@ class LaunchActivity : AppCompatActivity() {
      */
     private fun _checkForVersionUpdate(): Boolean {
         val savedVerCode = _customSettings.getInt(
-            getString(R.string.preference_key_version_code), 0)
+            getString(R.string.pref_key_version_code), 0)
         val curVerCode = try {
             packageManager.getPackageInfoCompat(packageName, 0).versionCodeCompat
         } catch (e: Exception) {
             0
         }
         return if (curVerCode > savedVerCode) {
             _customSettings.edit().apply {
-                putInt(getString(R.string.preference_key_version_code), curVerCode)
+                putInt(getString(R.string.pref_key_version_code), curVerCode)
             }.apply()
             true
         } else {
             false
         }
     }
+
+    /**
+     * Check whether the settings need to be migrated or not.
+     */
+    private fun _checkForSettingsMigration() {
+        if (_customSettings.getInt(getString(R.string.pref_key_pref_version), 1) == 1) {
```

Comment:
Wouldn't it make more sense to check for the latest version? Otherwise, we would always need to remember adapting this line when we increase the version.

Reply:
Duplicated comment.